### PR TITLE
Migrate To Modularization

### PR DESCRIPTION
**Pull request by hossein69ameri**
Review For Modularization

### CODE REVIEW

**Comment by hossein69ameri** [comment]
Rebase develop onto modularization

File: data/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Kaaveh) [inline data/src/main/AndroidManifest.xml:1]: Remove this unused file

File: domain/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+
+    implementation(libs.androidx.core.ktx)
+    implementation(project(":data"))
```
Comment (Kaaveh) [inline domain/build.gradle.kts]: domain shouldn't depend on the data module.

File: domain/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Kaaveh) [inline domain/src/main/AndroidManifest.xml:1]: Remove this unused file

File: presentation/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Kaaveh) [inline presentation/src/main/AndroidManifest.xml:1]: Remove this unused file

**Review by Kaaveh — CHANGES_REQUESTED** [review]

File: domain/src/main/java/com/ameri/domain/repository/HomeRepository.kt [hunk 5]
```diff
@@ -1,6 +1,6 @@
-package com.ameri.newsapi.data.repository
+package com.ameri.domain.repository
 
-import com.ameri.newsapi.data.network.ApiServices
+import com.ameri.data.network.ApiServices
 import javax.inject.Inject
 
 class HomeRepository @Inject constructor(private  val apiServices: ApiServices) {
```
Comment (Kaaveh) [inline domain/src/main/java/com/ameri/domain/repository/HomeRepository.kt]: Put this class into the data module.

Comment (Kaaveh) [inline domain/src/main/java/com/ameri/domain/repository/HomeRepository.kt]: Rename HomeRepository -> NewsRepository

Comment (Kaaveh) [inline domain/src/main/java/com/ameri/domain/repository/HomeRepository.kt]: Create an interface for this class and put that interface into the domain module (Checkout the ComposeNews's repositories)

File: data/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+
+    implementation(libs.androidx.core.ktx)
+    implementation(project(":app"))
```
Comment (Kaaveh) [inline data/build.gradle.kts]: The data module shouldn't depend on the app module

**Comment by hossein69ameri** [comment]
Please Check RuntimeError  : java.lang.RuntimeException: Cannot create an instance of class com.ameri.presentation.viewmodel.NewsViewModel

**Comment by hossein69ameri** [comment]
Restore Libraries and MinSDK 23

**Comment by hossein69ameri** [comment]
Fix DI Crash With Implement data Module in App Module and Add KSP To Gradle Plugin

**Review by Kaaveh — APPROVED** [review]

File: presentation/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```
Comment (Kaaveh) [inline presentation/src/main/AndroidManifest.xml]: Is this permission necessary?

File: data/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```
Comment (Kaaveh) [inline data/src/main/AndroidManifest.xml]: Is this permission necessary?

**Comment by hossein69ameri** [comment]
Fix Unused Code in Manifest and Unused Dependencies in Gradle

**Pull request merged**